src/Umbraco.Web.UI.Client/src/views/propertyeditors/grid/grid.controller.js

```diff
@@ -343,6 +343,37 @@ angular.module("umbraco")
             }
         };
 
+        var shouldApply = function(item, itemType, gridItem) {
+            if (item.applyTo === undefined || item.applyTo === null || item.applyTo === "") {
+                return true;
+            }
+
+            if (typeof (item.applyTo) === "string") {
+                return item.applyTo === itemType;
+            }
+
+            if (itemType === "row") {
+                if (item.applyTo.row === undefined) {
+                    return false;
+                }
+                if (item.applyTo.row === null || item.applyTo.row === "") {
+                    return true;
+                }
+                var rows = item.applyTo.row.split(',');
+                return _.indexOf(rows, gridItem.name) !== -1;
```

Comment:
If the row property is not null or empty (checked above) .. we will now split it by comma and see if any of the parts we get in the split array matches exactly on the `gridItem.name`. indexOf returns -1 if it doesnt exist in the array .. so that would evaluate to `return false` ... if indexOf returns anything but -1, it means it found it in the array and it will evaluate to `return true`.